File: examples/playwright_exemplo.mjs
// Exemplo Playwright (Node): abre uma página e imprime o título.
//   node /opt/hermes-examples/playwright_exemplo.mjs [url]
import { createRequire } from "node:module";

// ESM ignora NODE_PATH; o require resolve o playwright instalado globalmente.
const { chromium } = createRequire(import.meta.url)("playwright");

const url = process.argv[2] ?? "https://example.com";
const browser = await chromium.launch();
const page = await browser.newPage();
await page.goto(url);
console.log(await page.title());
await browser.close();

File: examples/playwright_exemplo.py
"""Exemplo Playwright (Python): abre uma página e imprime o título.

    python3 /opt/hermes-examples/playwright_exemplo.py [url]
    xvfb-run -a python3 /opt/hermes-examples/playwright_exemplo.py --headed
"""
import sys

from playwright.sync_api import sync_playwright


def main() -> None:
    args = [a for a in sys.argv[1:] if not a.startswith("--")]
    url = args[0] if args else "https://example.com"
    headed = "--headed" in sys.argv
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=not headed)
        page = browser.new_page()
        page.goto(url)
        print(page.title())
        browser.close()


if __name__ == "__main__":
    main()
